python/ray/air/tests/test_checkpoints.py

```diff
@@ -728,6 +728,60 @@ def testDirCheckpointWithPreprocessorAsDict(self):
             preprocessor = checkpoint.get_preprocessor()
             assert preprocessor.multiplier == 1
 
+    def testDictCheckpointSetPreprocessor(self):
+        preprocessor = DummyPreprocessor(1)
+        data = {"metric": 5}
+        checkpoint = Checkpoint.from_dict(data)
+        checkpoint.set_preprocessor(preprocessor)
+        preprocessor = checkpoint.get_preprocessor()
+        assert preprocessor.multiplier == 1
+
+    def testDictCheckpointSetPreprocessorAsDir(self):
+        preprocessor = DummyPreprocessor(1)
+        data = {"metric": 5}
+        checkpoint = Checkpoint.from_dict(data)
+        checkpoint.set_preprocessor(preprocessor)
+        checkpoint_path = checkpoint.to_directory()
+        checkpoint = Checkpoint.from_directory(checkpoint_path)
+        preprocessor = checkpoint.get_preprocessor()
+        assert preprocessor.multiplier == 1
+
+    def testDirCheckpointSetPreprocessor(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            preprocessor = DummyPreprocessor(1)
+            data = {"metric": 5}
+            checkpoint_dir = os.path.join(tmpdir, "existing_checkpoint")
+            os.mkdir(checkpoint_dir, 0o755)
+            with open(os.path.join(checkpoint_dir, "test_data.pkl"), "wb") as fp:
+                pickle.dump(data, fp)
+            checkpoint = Checkpoint.from_directory(checkpoint_dir)
+            checkpoint.set_preprocessor(preprocessor)
+            preprocessor = checkpoint.get_preprocessor()
+            assert preprocessor.multiplier == 1
+
+    def testDirCheckpointSetPreprocessorAsDict(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            preprocessor = DummyPreprocessor(1)
+            data = {"metric": 5}
+            checkpoint_dir = os.path.join(tmpdir, "existing_checkpoint")
+            os.mkdir(checkpoint_dir, 0o755)
+            with open(os.path.join(checkpoint_dir, "test_data.pkl"), "wb") as fp:
+                pickle.dump(data, fp)
+            checkpoint = Checkpoint.from_directory(checkpoint_dir)
+            checkpoint.set_preprocessor(preprocessor)
+            checkpoint_dict = checkpoint.to_dict()
+            checkpoint = checkpoint.from_dict(checkpoint_dict)
+            preprocessor = checkpoint.get_preprocessor()
+            assert preprocessor.multiplier == 1
+
+    def testObjectRefCheckpointSetPreprocessor(self):
+        ckpt = Checkpoint.from_dict({"x": 1})
+        ckpt = ray.get(ray.put(ckpt))
```

Comment:
I don't think this `ckpt` is an object ref checkpoint. When I checked`ckpt._obj_ref` here, I got `None`. 

Is this what we want?

```
obj_ref = ray.put({"x": 1})
ckpt = Checkpoint.from_object_ref(obj_ref)
```

Reply:
`from_object_ref` is deprecated in favor of `ray.get`?

Seems like the `_obj_ref` path is not used anymore at all.

Reply:
Ah, yeah, you're right. I'll open a PR to remove the dead code.